src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -780,7 +780,7 @@ static inline bool notification_match(reservation_t& res,
   const RGWPubSub ps(res.store, res.user_tenant);
   const RGWPubSub::Bucket ps_bucket(ps, res.bucket);
   rgw_pubsub_bucket_topics bucket_topics;
-  auto rc = ps_bucket.get_topics(res.dpp, bucket_topics, res.yield);
+  auto rc = ps_bucket.get_topics(res.dpp, bucket_topics, res.s->yield);
```

Comment:
i think this is the cause of the crash you're seeing in teuthology. `RGWDeleteMultiObj::handle_individual_object()` runs under a different yield context than the req_state's `res.s->yield`